### PR TITLE
Add methods for opcontrol on the chassis

### PR DESCRIPTION
**Pull request by BattleCh1cken**
#### Summary

This pull request adds methods to the chassis that allow the robot to controlled during opcontrol.

Currently three different move methods are planned:

- tank drive
- arcade drive
- curvature drive

All of these movement methods can be modified with a drive curve

#### Test Plan

Upload the code to a robot, and drive around.

#### Additional Notes

This PR is obviously incomplete, and requires some work. This is because I wanted to ask if I should add support for the following features that I've seen mentioned:
- deadzones
- negative inertia accumulator (mentioned [here](https://wiki.purduesigbots.com/software/competition-specific/curvature-cheesy-drive))
- active braking

### CODE REVIEW

**Comment by BattleCh1cken** [comment]
As of now the PR is tested and could be used as is. If the other features are desired I can still add them.

File: include/lemlib/chassis/chassis.hpp [hunk 1]
```diff
@@ -164,6 +164,39 @@ class Chassis {
...
                     bool log = false);
+        /**
+         * @brief Control the robot during the driver control period using the tank drive control scheme. In this
+         * control scheme one joystick axis controls one half of the robot, and another joystick axis controls another.
+         * @param left speed of the left side of the drivetrain. Takes an input from -127 to 127.
+         * @param right speed of the right side of the drivetrain. Takes an input from -127 to 127.
+         * @param curveGain control how steep the drive curve is. The larger the number, the steeper the curve. A value
+         * of 1 disables the curve entirely.
+         */
+
```
Comment (liam-teale) [inline include/lemlib/chassis/chassis.hpp]: extra whitespace. How dare you?

File: include/lemlib/chassis/chassis.hpp [hunk 2]
```diff
@@ -164,6 +164,39 @@ class Chassis {
...
+         * @brief Control the robot during the driver using the curvature drive control scheme. This control scheme is
+         * very similar to arcade drive, except the second joystick axis controls the radius of the curve that the
+         * drivetrain makes, rather than the speed. This means that the driver can accelerate in a turn without changing
+         * the radius of that turn. This control scheme defaults to arcade when forward is zero.
+         * @param forward speed to move forward. Takes an input from -127 to 127.
+         * @param turn speed to turn. Takes an input from -127 to 127.
+         * @param curveGain control how steep the drive curve is. The larger the number, the steeper the curve. A value
+         * of 1 disables the curve entirely.
+         */
+
```
Comment (liam-teale) [inline include/lemlib/chassis/chassis.hpp]: :sob:

File: src/lemlib/chassis/opcontrol.cpp [hunk 3]
```diff
@@ -0,0 +1,43 @@
+#include "lemlib/chassis/chassis.hpp"
+#include <algorithm>
+#include <math.h>
+
+namespace lemlib {
+
+double calcDriveCurve(double input, double scale) {
```
Comment (liam-teale) [inline src/lemlib/chassis/opcontrol.cpp:14]: Missing doxygen comment

File: src/lemlib/chassis/opcontrol.cpp [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+    if (scale != 0) {
+        return (powf(2.718, -(scale / 10)) + powf(2.718, (fabs(input) - 127) / 10) * (1 - powf(2.718, -(scale / 10)))) *
+               input;
+    }
+    return input;
+}
+
+void Chassis::tank(int left, int right, float curveGain) {
```
Comment (liam-teale) [inline src/lemlib/chassis/opcontrol.cpp:31]: Missing doxygen comment

File: src/lemlib/chassis/opcontrol.cpp [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+    drivetrain.leftMotors->move(calcDriveCurve(left, curveGain));
+    drivetrain.rightMotors->move(calcDriveCurve(right, curveGain));
+};
+
+void Chassis::arcade(int forward, int turn, float curveGain) {
```
Comment (liam-teale) [inline src/lemlib/chassis/opcontrol.cpp]: Missing doxygen comment

File: src/lemlib/chassis/opcontrol.cpp [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+    int leftPower = calcDriveCurve(forward + turn, curveGain);
+    int rightPower = calcDriveCurve(forward - turn, curveGain);
+    drivetrain.leftMotors->move(leftPower);
+    drivetrain.rightMotors->move(rightPower);
+};
+
+void Chassis::curvature(int forward, int curvature, float cureveGain) {
```
Comment (liam-teale) [inline src/lemlib/chassis/opcontrol.cpp]: Missing doxygen comment

**Review by liam-teale — CHANGES_REQUESTED** [review]

**Review by liam-teale — APPROVED** [review]

**Pull request merged**